Doc/Doc/Perldl.pm

```diff
@@ -77,7 +86,7 @@ sub screen_width() {
     return $ENV{COLUMNS}
        || (($ENV{TERMCAP} =~ /co#(\d+)/) and $1)
        || ($^O ne 'MSWin32' and $^O ne 'dos' and 
-	   (`stty -a 2>/dev/null` =~ /columns\s*=?\s*(\d+)/) and $1)
+	  ((`stty -a 2>/dev/null` =~ /columns\s*=?\s*(\d+)/) or (`stty -a 2>/dev/null` =~ /(\d+)\s+columns/)) and $1)
```

Comment:
Is this not encapsulated in a CPAN module?

Reply:
Yes.  [perlfaq8](https://perldoc.perl.org/5.30.0/perlfaq8.html#How-do-I-get-the-screen-size%3f) suggests using [GetTerminalSize](https://metacpan.org/source/JSTOWE/TermReadKey-2.38/ReadKey.pm.PL#L399) in [Term::ReadKey](https://metacpan.org/pod/Term::ReadKey), and we could add that as a CPAN dependency since Term::ReadKey is not a Perl core module.

FWIW that function, after some handling of more esoteric systems (which may include Windows 😕) , seems to look for either
- $ENV{COLUMNS}
- the `resize` command (comes with X11 or other window managers), or
- `stty size` (which may be a cleaner than `stty -a`)

If foisting the cross-platform compatibility issues onto somebody else is worth adding a dependency, then that's fine with me.

Reply:
If we need to get this information, best not to try to do it ourselves - PDL is about whizzy data, not about terminal-handling :-)

Derek, do you feel like also switching this code to use Term::ReadKey? I'm on the verge of full-releasing PDL, in a few days, and would like to then be able to merge this and push out an immediate dev-release with it, for people to try.

Reply:
Sure, that's fine.  I cooked up a Term::ReadKey -optional version but then since it was so easy to install from CPAN I just got rid of all of our old code.  For my own reference, that half-and-half version looked like:

~~~~perl
sub screen_width() {
    local $@;
    eval { require Term::ReadKey; };
    unless ( $@ ){
	return ( ( Term::ReadKey::GetTerminalSize(\*STDOUT) )[0] );
    } else {
    return $ENV{COLUMNS}
       || (($ENV{TERMCAP} =~ /co#(\d+)/) and $1)
       || ($^O ne 'MSWin32' and $^O ne 'dos' and
	  ((`stty -a 2>/dev/null` =~ /columns\s*=?\s*(\d+)/) or (`stty -a 2>/dev/null` =~ /(\d+)\s+columns/)) and $1)
       || 72;
    }
}
~~~~
But now it's just
~~~~perl
use Term::ReadKey
sub screen_width() {
    return ( ( GetTerminalSize(\*STDOUT) )[0] );
}
~~~~
My only concern with this approach is that somebody who doesn't have Term::ReadKey already installed will have a broken perldl shell install if they do an upgrade and don't see the one-line warning about not having the new prerequisite.  Should Makefile.PL die when those prerequisites aren't satisfied?